R/utils.R

```diff
@@ -207,53 +214,85 @@ cache_data <- function(region = c("nwfsc", "pbs", "afsc")) {
 load_sql_data <- function() {
   f <- files_to_cache()
 
+  # Detect if running in CI
+  in_ci <- Sys.getenv("GITHUB_ACTIONS") == "true"
+
   f_haul <- sort(f[grepl("haul", f)])
   f_catch <- sort(f[grepl("catch", f)])
-  stopifnot(length(f_haul) == length(f_catch))
-  haul <- map_dfr(f_haul, function(x) {
-    # error handling if file doesn't exist -- largely for CI on Github
-    this_file <- file.path(get_cache_folder(), x)
-    if (!file.exists(this_file)) {
-      cli_inform(paste("File does not exist: ", this_file))
-      out <- NULL
+
+  # read and clean data
+  read_rds_file <- function(x) {
+    if (in_ci) {
+      # run in CI: download from GitHub w/o saving
+      url <- paste0("https://raw.githubusercontent.com/DFO-NOAA-Pacific/surveyjoin-data/main/", x)
+      cli::cli_alert_info("Reading remote file from GitHub: {url}")
+      # create tempfile
+      temp_file <- tempfile(fileext = ".rds")
+
+      result <- tryCatch({
+        # Download to temp file and read in file
+        download.file(url, destfile = temp_file, mode = "wb", quiet = TRUE)
+        data <- readRDS(temp_file)
+        return(data)
+      }, error = function(e) {
+        cli::cli_alert_warning("Failed to download {x}")
+        return(NULL)
+      }, finally = {
+        # delete temp file
+        if (file.exists(temp_file)) unlink(temp_file, recursive = TRUE, force = TRUE)
+      })
+      return(result)
     } else {
-      out <- readRDS(file.path(get_cache_folder(), x))
-      out$region <- gsub("([a-z]+)-[a-z]+.rds", "\\1", x)
-      if (is.character(out$event_id)) out$event_id <- as.numeric(out$event_id)
-      # out$event_id <- as.character(out$event_id)
-      out$performance <- as.character(out$performance)
-      out$year <- as.integer(lubridate::year(out$date))
-      out$date <- as.character(lubridate::as_date(out$date))
-      # FIXME: do this long before! Alaska
-      out$lon_start <- ifelse(out$lon_start > 0, out$lon_start * -1, out$lon_start)
-      out$lon_end <- ifelse(out$lon_end > 0, out$lon_end * -1, out$lon_end)
-      if (out$region[1] == "pbs") {
-        # Fix me earlier!
-        out <- dplyr::rename(out, bottom_temp_c = .data$temperature_C) %>%
-          dplyr::select(-.data$do_mlpL, -.data$salinity_PSU)
+      # Read from local cache
+      local_file <- file.path(get_cache_folder(), x)
+      if (!file.exists(local_file)) {
+        cli::cli_abort("File missing locally: {local_file}")
```

Comment:
I am unsure if we should format this as a file name because the file doesn't exist, e.g.,
```suggestion
        cli::cli_abort("File missing locally: {.file {local_file}}")
```